lib/ansible/modules/cloud/amazon/elasticache_parameter_group.py

```diff
@@ -0,0 +1,237 @@
+
+
+"""
+Elasticache parameter groups module functionality
+"""
+
+
+def upsert_elasticache_pg(es_client, existing_pg, module):
+
+    try:
+        response = existing_pg
+        changed = False
+        if not existing_pg:
+            response = es_client.create_cache_parameter_group(
+                CacheParameterGroupName=module.params.get("pg_name"),
+                CacheParameterGroupFamily=module.params.get("pg_family"),
+                Description=module.params.get("pg_desc")
+            )["CacheParameterGroup"]
+            changed = True
+
+        is_param_changes = False
+        if module.params.get("pg_params"):
+            response["params"] = get_current_pg_params(es_client, module)
+            is_param_changes = is_params_diff(
+                response["params"], module.params.get("pg_params"))
+            response = es_client.modify_cache_parameter_group(
+                CacheParameterGroupName=module.params.get("pg_name"),
+                ParameterNameValues=convert_params_format(
+                    module.params.get("pg_params"))
+            )
+            if is_param_changes:
+                changed = True
+            response["params"] = dict([
+                (x["ParameterName"], x["ParameterValue"])
+                for x in get_current_pg_params(es_client, module)])
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg=e.message)
+
+    return response, changed
+
+
+def remove_elasticache_pg(es_client, existing_pg, module):
+
+    try:
+        if existing_pg:
+            response = es_client.delete_cache_parameter_group(
+                CacheParameterGroupName=module.params.get("pg_name")
+            )
+            changed = True
+        else:
+            response = "Parameter Group doesn't exist, not doing anything"
+            changed = False
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg=e.message)
+
+    return response, changed
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+        pg_name=dict(required=True),
+        pg_desc=dict(),
+        pg_family=dict(choices=[
+            "memcached1.4", "redis2.6", "redis2.8", "redis3.2"]),
+        pg_params=dict(type="dict"),
+        state=dict(required=True, choices=["absent", "present"])))
+
+    module = AnsibleModule(
+        argument_spec=argument_spec
+    )
+
+    if not HAS_BOTO3:
+        module.fail_json(msg="boto3 required for this module")
+
+    state = module.params.get("state")
+    es_client = get_es_client(module)
+    existing_pg = get_current_pg(es_client, module)
+
+    if state == "absent":
+        response, changed = remove_elasticache_pg(
+            es_client, existing_pg, module)
+    elif state == "present":
+        response, changed = upsert_elasticache_pg(
+            es_client, existing_pg, module)
+
+    module.exit_json(changed=changed, parameter_groups=response)
+
+from ansible.module_utils.basic import *
```

Comment:
I put mine after the comments just above the documentation variable as then it passes pep8.